src/tty-ask-password-agent/tty-ask-password-agent.c

```diff
@@ -209,6 +228,72 @@ static int ask_password_plymouth(
         return 0;
 }
 
+static void free_consoles(struct console *con, unsigned int num) {
+        unsigned int n;
+        if (!con || !num)
+                return;
+        for (n = 0; n < num; n++)
+                free(con[n].tty);
+        free(con);
+}
+
+static const char *current_dev = "/dev/console";
+static int collect_consoles(struct console **consoles, unsigned int *num) {
+        _cleanup_free_ char *active = NULL;
+        const char *word, *state;
+        struct console *con = NULL;
+        size_t con_len = 0, len;
+        unsigned int count = 0;
+        int ret;
+
+        ret = read_one_line_file("/sys/class/tty/console/active", &active);
+        if (ret < 0)
+                return ret;
+        FOREACH_WORD(word, len, active, state) {
```

Comment:
Should the pointer declaration moved more into the head of tty-ask-password-agent.c?